packages/colorpickers/src/elements/ColorPicker/ColorWell.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useCallback, useContext, useEffect, useMemo } from 'react';
+import throttle from 'lodash.throttle';
+import { ThemeContext } from 'styled-components';
+import { useDocument } from '@zendeskgarden/react-theming';
+import { IHSVColor } from '../../utils/types';
+import { hslToHsv } from '../../utils/conversion';
+import { getNextHsv, getThumbPosition } from '../../utils/saturation';
+import {
+  StyledColorWell,
+  StyledColorWellGradient,
+  StyledColorWellThumb,
+  StyledColorWellWrapper
+} from '../../styled';
+
+interface IColorWellProps {
+  hue: number;
+  saturation: number;
+  lightness: number;
+  onChange?: (hsv: IHSVColor, event: MouseEvent) => void;
+}
+
+export const ColorWell: React.FC<IColorWellProps> = ({ hue, saturation, lightness, onChange }) => {
+  const theme = useContext(ThemeContext);
+  const environment = useDocument(theme);
+  const { rtl } = useContext(ThemeContext);
+  const container = useRef<HTMLDivElement>(null);
```

Comment:
This should be named `containerRef` to follow naming conventions. This also allows the rules-of-hooks to understand dependencies better.